user/serializers.py

```diff
@@ -1 +1,24 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+from user.models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "email", "password", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {"password": {"write_only": True, "min_length": 5}}
+
+    def create(self, validated_data: dict) -> User:
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance: User, validated_data: dict) -> User:
+        password = validated_data.pop("password", None)
+        user = super().update(instance, validated_data)
+        if password:
+            user.set_password(password)
+            user.save()
+        return user
```

Comment:
it's already implemented in ModelSerializer

Reply:
I'm sorry, but i didn't find where it is implemented in ModelSerializer. I updated create and update method, as shown in lessons,  because it needed to implement create user (hashed user password).